libazureinit/src/config.rs

```diff
@@ -878,6 +1097,22 @@ mod tests {
         path = "/cli-override-azure-init-data-dir"
         [azure_init_log_path]
         path = "/custom/path/azure-init.log"
+        [runcmd]
+        commands = [
+            ["echo", "hello"],
+            ["apt-get", "update"]
+        ]
+        fail_provisioning_upon_failure = false
```

Comment:
can we add a sample and test case for `gz+b64`? Also add a test for multi-line content